### ISSUE TITLE
doctest fails with Octave 4.3+

### ISSUE THREAD

**Issue by mtmiller** [cbm755/octsympy#777]
A handful of doctests are failing in Octave 4.3+, specifically due to a change in the display of cell arrays containing objects.

In Octave 4.2 and earlier

    >> x{1} = sym ("x")
    x = 
    {
    (sym) x
    }

In Octave 4.3+, this has now become

    >> x{1} = sym ("x")
    x = 
    {
      [1,1] =
    
        <class sym>
    }

Can someone show what a cell array of objects ought to look like in a recent version of Matlab? Then we can see whether and how to address this?

**Comment by mtmiller** [comment]
I filed [#50756](https://savannah.gnu.org/bugs/?50756) upstream, comments welcome either here or there.

**Comment by cbm755** [comment]
IIRC, Matlab does not display complicated objects within cell-arrays, but I cannot easily check at the moment.

If we really want to properly support printing inside cell arrays, the display method needs to be made aware of the current indentation level.  IIRC, @oheim was interested in this.  There may be other upstream bugs about it too.

**Comment by nvs-abhilash** [comment]
This is the output in MATLAB 2017a

```
>> x{1} = sym("x")

x =

  cell

    [1×1 sym]

```

**Comment by latot** [comment]
i really don't like don't show the content of cells..

**Comment by mtmiller** [comment]
There is some discussion upstream to bring back displaying the contents of cell array entries.

There are also other upstream bugs about changing the display of cell arrays entirely. It seems like something will change, but we may have to wait and see which direction Octave chooses. And even then the doc strings may depend on the version of Octave.

**Comment by cbm755** [comment]
My working idea is that doctests should target the most current release.  This is b/c they are primarily documentation.  So no huge hurry to make our doctests pass on the daily hg master Octave.

@mtmiller: this could be extended as an argument for not running doctest with `OCT=daily` on CI?

**Comment by mtmiller** [comment]
> My working idea is that doctests should target the most current release

With the added delay of Travis updating its image, and Octave releases being packaged for Debian and Ubuntu. Ok, I'm good with that if you are.

I agree, we could disable doctest on the daily build.

You can close this if you are saying it's premature here, and we can decide upstream what the cell format will end up looking like.

**Comment by mtmiller** [comment]
I think it's likely the cell array format bug will not be resolved by the time Octave 4.4 is stamped.

With that in mind, from a different angle, there are only three methods and one function that fail in Octave 4.4 because of the change to the cell array output format: sym/children, sym/dsolve, sym/symreplace, and findsymbols. Maybe it would be better to rewrite the doc strings so that they don't depend on the appearance of cell arrays?

Something like this for `@sym/children.m`:

```
%% @example
%% @group
%% A = [x*y 2; 3 x]
%%   @result{} A = (sym 2×2 matrix)
%%       ⎡x⋅y  2⎤
%%       ⎢      ⎥
%%       ⎣ 3   x⎦
%% C = children(A);
%% C@{:@}
%%   @result{} ans = (sym) [x  y]  (1×2 matrix)
%%     ans = (sym) 3
%%     ans = (sym) 2
%%     ans = (sym) x
%% size(C)
%%   @result{} ans =
%%        2   2
%% @end group
%% @end example
```

passes in both Octave 4.2 and 4.4.

There is one more doctest failure in Octave 4.4 that is not due to cell array format:

```
laguerreL .............................................. FAIL    1/4   

   >> ans - double (exact)

      expected:    -1.6798e-11

      got     : ans = -0.000000000016798
```

**Comment by cbm755** [comment]
Thanks, I'll make those changes before next release and look into the laguerreL.
